File: lume/lume/lume/Domain/Notifica.cs
using System;
using System.Text.Json.Serialization;
using lume.Domain;

namespace lume.Domain
{
    public class Notifica
    {

        [JsonPropertyName("id")]
        public long id { get; set; }


        [JsonPropertyName("titolo")]
        public string titolo { get; set; }


        [JsonPropertyName("descrizione")]
        public string descrizione { get; set; }


        [JsonPropertyName("richiesta")]
        public Richiesta richiesta { get; set; }


        [JsonPropertyName("utente")]
        public Utente utente { get; set; }

        [JsonPropertyName("soggetto")]
        public Utente soggetto { get; set; }
    }
}

File: API/src/main/java/it/lume/mc/model/Notifica.java
package it.lume.mc.model;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.Data;

@Entity
public @Data class Notifica {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	
	private String descrizione;
	
	@JsonIgnoreProperties(value = {"nome", "cognome", "email", "telefono", "notifiche", "richiestePartecipate", "richieste"})
	@ManyToOne
	private Utente utente;
	
	@JsonIgnoreProperties(value = {"nome", "cognome", "email", "telefono", "notifiche", "richiestePartecipate", "richieste"})
	@OneToOne
	private Utente soggetto;

	@JsonIgnoreProperties(value = {"titolo", "descrizione", "creatore", "candidati", "notifiche", "numeroPartecipanti", "dataCreazione"})
	@ManyToOne
	private Richiesta richiesta;

}
